config/components/prometheus/monitor_tls_patch.yaml

```diff
@@ -0,0 +1,19 @@
+# Patch for Prometheus ServiceMonitor to enable secure TLS configuration
+# using certificates managed by cert-manager
+- op: replace
+  path: /spec/endpoints/0/tlsConfig
+  value:
+    # SERVICE_NAME and SERVICE_NAMESPACE will be substituted by kustomize
+    serverName: SERVICE_NAME.SERVICE_NAMESPACE.svc
+    insecureSkipVerify: false
+    ca:
+      secret:
+        name: metrics-server-cert
+        key: ca.crt
+    cert:
+      secret:
+        name: metrics-server-cert
```

Comment:
IMO it is a bit weird to let the prometheus client use the same cert that we use in our server. I have left this to the user to specify the certificate their prometheus instance is using https://github.com/kubernetes-sigs/jobset/pull/863/files#diff-d1323215bef44ba29d323d647a042eed78e05ce849be1e3f8c067a89fd735359R23-R26

Reply:
We are hardcoding nearly all fields and in my opinion we can hardcode it too.

Reply:
IMO it is not entirely correct to let the prometheus use this as a client cert for auth. As mentioned in https://github.com/kubernetes-sigs/jobset/pull/863#discussion_r2032082412 there are many things that have to be configured before running cert manager + prometheus. So I think the best way would be to let user decide if they want to use the server certificate for this. They can easily opt in to it.

I have tried to document this in:
- https://github.com/atiratree/kubernetes-sigs-jobset/blob/enable-tls-metrics/config/components/prometheus/kustomization.yaml
- https://github.com/atiratree/kubernetes-sigs-jobset/blob/enable-tls-metrics/config/components/prometheus/monitor_tls_patch.yaml

Reply:
> there are many things that have to be configured before running cert manager + prometheus.

We only require to uncomment some configurations in kustomize. 

If user relies on cert-manager, we can use the same certificates both in client and server because source of truth is the same (i.e. cert-manager) within this cluster.

If user relies on internal certificate management, skipping tls verification in servicemonitor would simply work. If we force users to find and inject a certificate in servicemonitor, users can't inject any arbitrary certificate. This certificate must comply with the one that is managed internally in LWS. I think, this dramatically hurts the user experience. Because I don't think, as a user I can successfully configure this cluster on my own.

Reply:
I understand the usability perspective, but technically that certificate is wrong. In mutual TLS the domain name would be off.

I am okay, by adding it as a default, but IMO it should be configurable and we should write a disclaimer there.